Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h

```diff
@@ -3134,14 +3134,15 @@ bool AbstractInterpreter<AbstractStateType>::executeEffects(unsigned clobberLimi
                 structure = globalObject->nullPrototypeObjectStructure();
             else if (base.isObject()) {
                 // Having a bad time clears the structureCache, and so it should invalidate this structure.
-                bool isHavingABadTime = globalObject->isHavingABadTime();
-                // Normally, we would always install a watchpoint. In this case, however, if we haveABadTime, we
+                bool isStructureCacheCleared = globalObject->isStructureCacheCleared();
+                // Normally, we would always install a watchpoint. In this case, however, if we cleared StructureCache, we
                 // still want to optimize. There is no watchpoint for that case though, so we need to make sure this load
```

Comment:
Is this comment still correct?  Do we "still want to optimize" here if we know that the structureCache is cleared.  Your code change below suggests otherwise.

Reply:
This comment is no longer valid, let's just drop it.